learning/step_by_step/ui_code_a_life_bar.rst

```diff
@@ -0,0 +1,590 @@
+its own scripts, and so should the UI elements. This prevents bugs,
+keeps your project manageable, and allows different team members to work
+on different parts of the game.
+
+Once the core gameplay and the UI are ready, you'll need to connect them
+somehow. In our example, we have the Enemy who attacks the Player at
+constant time intervals. We want the life bar to update when the Player
+takes damage.
+
+To do this, we will use **signals**.
+
+.. note::
+
+::
+
+    Signals are Godot's version of the Observer pattern. They allow us to send out some message. Other nodes can connect to the object that **emits** the signal and receive the information. It's a powerful tool we use a lot for User Interface and achievement systems. You don't want to use them everywhere though. Connecting two nodes adds some coupling between them. When there's a lot of connections, they become hard to manage.
+    For more information on check out the [ signals video tutorial ](https://youtu.be/l0BkQxF7X3E) on GDquest.
+
+Download and explore the start project
+--------------------------------------
+
+Download the Godot project. It contains all the assets and scripts you
+need to get started. Extract the .zip archive to get two folders:
+
+-  tutorial/ contains the start and end godot games for this tutorial
+-  exercise/ lets you practice what you'll learn here by adding a new
+   GUI element on your own. You'll find more on that at the bottom of
+   the page
+
+Load the ``tutorial/start`` project in Godot. In the ``FileSystem`` dock
+double click on LevelMockup.tscn to open it. It's an RPG game's mockup
+where 2 characters face each other. The pink enemy attacks and damages
+the green square at regular time intervals, until its death. Feel free
+to try out the game: the basic combat mechanics already work. But as the
+character isn't connected to the life bar the ``GUI`` doesn't do
+anything.
+
+.. note::
+
+::
+
+    This is typical of how you'd code a game: you implement the core gameplay first, handle the player's death, and only then you'll add the interface. That's because the UI listens to what's happening in the game. So it can't work if other systems aren't in place yet.
+    If you design the UI before you prototype and test the gameplay, chances are it won't work well and you'll have to re-create it from scratch.
+
+The scene contains a background sprite, a GUI, and two characters.
+
+.. figure:: img/life_bar_step_tut_LevelMockup_scene_tree.png
+   :alt: The scene tree, with the GUI scene set to display its children
+
+   The scene tree, with the GUI scene set to display its children
+
+The GUI scene encapsulates all of the Game User Interface. It comes with
+a barebones script where we get the path to nodes that exist inside the
+scene:
+
+::
+
+    onready var Number = $Bars/LifeBar/Count/Background/Number
+    onready var Bar = $Bars/LifeBar/TextureProgress
+    onready var Tween = $Tween
```

Comment:
Use lowercase for the variables, since you are currently shadowing the `Tween` class (and it is questionable if it would let you do that anyway). Also it is more GDScript-idiomatic that way.
So:
```gdscript
onready var number = $Bars/LifeBar/Count/Background/Number
onready var bar = $Bars/LifeBar/TextureProgress
onready var tween = $Tween
```

Also, `number` is a bit vague, maybe `number_label` or just `label`?

Reply:
Okay, so even for objects, always lowercase vars in GDscript? I tend to use PascalCase because you get that by default in the editor, so treating each of them like a class instance. Anyway that's fine by me. Gonna fix it